carma-messenger-core/cpp_message/src/MobilityPath_Message.cpp

```diff
@@ -68,10 +69,15 @@ namespace cpp_message
             header.recipient_id=recipient_id;
             
             //get bsm id
+            //sender_bsm_id is meant to represent the vehicle BSM id in hex string (Ex: FFFFFFFF)
+            //Since we're decoding this from a binary value to a MobilityHeader value, the transition would be from Binary to Hex
             str_len=message->value.choice.TestMessage02.header.hostBSMId.size;
+            std::string temp;
             for(size_t i=0;i<str_len;i++){
-                sender_bsm_id +=message->value.choice.TestMessage02.header.hostBSMId.buf[i];
+                temp +=message->value.choice.TestMessage02.header.hostBSMId.buf[i];
```

Comment:
Based on this the TestMessage03 change earlier was probably just a copy paste error